partner_compassion/models/partner_compassion.py

```diff
@@ -528,18 +528,21 @@ def forget_me(self):
                 "geo_point": False,
                 "partner_latitude": False,
                 "partner_longitude": False,
+                "birthdate_date": False,
             }
         )
-        self.advocate_details_id.unlink()
-        self.survey_inputs.unlink()
-        self.env["mail.tracking.email"].search([("partner_id", "=", self.id)]).unlink()
-        self.env["auditlog.log"].search(
+        self._cr.execute("""update res_partner set ref='', global_id='' where id=""" + str(self.id))
```

Comment:
This difference avoids SQL injection
```suggestion
        self._cr.execute("update res_partner set ref=NULL, global_id=NULL where id=%s", [self.id])
```